### DISCUSSION TITLE
"Error" during installation of Miele integration

### DISCUSSION THREAD

**Discussion by Stefan28L** [astrandb/miele#300] [Q&A] [answered]
I installed the integration by HACS.
I try to add the integration in HA which will start the configuration assistant.
I have already entered the miele developer information (ClientID etc.).
This takes me to the miele login page.
After i entered my login information it redirected me back to  HA.
There i get the simple message-window "Error" which i can close with a "OK" button.

Now, every time i try to add the integration, i will see the miele login page, No longer asking for the Client ID.
I'm shure the login information itself is correct.

I find an error in my log file. Here is a smale cutout of the message:

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Logger: aiohttp.server
Source: helpers/config_entry_oauth2_flow.py:209
First occurred: 15:49:27 (1 occurrences)
Last logged: 15:49:27

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
.
.
.
 File "/usr/src/homeassistant/homeassistant/helpers/config_entry_oauth2_flow.py", line 209, in _token_request
    resp.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1005, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 401, message='Unauthorized', url=URL('https://api.mcs3.miele.com/thirdparty/token')
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

**Comment by Stefan28L** [comment]
After I deleted the credentials in home assistant and re-entered them, the connection works now.

**Comment by trinksrichard** [comment]
@Stefan28L Where can you delete login details in Home Assistant? I entered the wrong token and cannot progress any further. I can't find that information anywhere.

**Comment by astrandb** [comment, marked as answer]
Please look here: https://github.com/astrandb/miele/discussions/247

**Comment by trinksrichard** [comment]
Thank you for your quick response.
Just found the right file, which I could edit with editor 'nano' using the command below

nano /config/.storage/application_credentials

**Reply by astrandb** [reply]
Be careful when you edit files in .storage/config . A mistake can easily crash your system.
